apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -299,30 +304,224 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
     test "find the definition when calling a erlang module", %{project: project} do
       subject_module = ~q[
         :erlang.binary_to_atom|("1")
-      ]
+        ]
       {:ok, uri, definition_line} = definition(project, subject_module)
 
       assert uri =~ "/src/erlang.erl"
       assert definition_line =~ ~S[«binary_to_atom»(Binary)]
     end
   end
 
-  defp caller_position(subject_module) do
-    {line, character} = cursor_position(subject_module)
-    Position.new(line, character)
+  describe "resolve/2" do
+    test "syntax error", %{project: project} do
+      code = ~q[
+        Some.Modul|e.
+        ]
+
+      assert {:error, _} = resolve(project, code)
```

Comment:
I originally wanted to support this case but was struggling to figure out how to handle it. I believe it should be possible, but I need to use `container_cursor_to_quoted` at the position of the syntax error, not the position of the cursor, when creating the fragment. I'm inclined to not support hover in the case of syntax errors to start (lexical immediately highlights the error anyways), and then add more robust entity resolution in a later PR.

Reply:
You might want to use `Code.Fragment.surround_context` to pull out the context instead of or in combination with `Code.Fragment.cursor_context`.